davinci-dome/src/DaVinci.ts

```diff
@@ -178,20 +177,26 @@ export function degreesToRadians(degree: number): number {
 export function daVinciOutput({bars, bolts, joints}: DaVinciResult): DaVinciOutput {
     const daVinciIntervals: DaVinciInterval [] = []
     bars
-        .filter(bar => bar.jointA.position === JointPosition.Above || bar.jointD.position === JointPosition.Above)
+        .filter(bar => bar.joints[0].position === JointPosition.Above || bar.joints[bar.joints.length - 1].position === JointPosition.Above)
         .forEach((bar) => {
             daVinciIntervals.push({
-                nodeIndexes: [bar.jointA.index, bar.jointB.index, bar.jointC.index, bar.jointD.index],
+                nodeIndexes: bar.joints.map(j => j.index),
                 type: "type 1",
             })
         })
     bolts
-        .filter(bolt => bolt.jointA.position === JointPosition.Above || bolt.jointB.position === JointPosition.Above)
+        .filter(bolt => bolt.jointA.position === JointPosition.Above && bolt.jointB.position === JointPosition.Above)
         .forEach((bolt) => {
-        daVinciIntervals.push({
-            nodeIndexes: [bolt.jointA.index, bolt.jointB.index],
-            type: "type 2",
+            daVinciIntervals.push({
+                nodeIndexes: [bolt.jointA.index, bolt.jointB.index],
+                type: "type 2",
+            })
         })
-    })
     return {joints, daVinciIntervals}
 }
+
+function sawBar(bar: Bar, planeJoint: Joint): Bar {
```

Comment:
Saw the bar means add the plane joint to either the front end or the back end (depending!), and then remove all joints with JointPosition.Below.